etc/kayobe/environments/ci-multinode/seed.yml

```
@@ -3,9 +3,7 @@ seed_bootstrap_user: "{{ os_distribution if os_distribution == 'ubuntu' else 'cl
 seed_lvm_groups:
   - "{{ stackhpc_lvm_group_rootvg }}"
 
-seed_extra_network_interfaces: >
-  "{{ seed_extra_network_interfaces_external +
-  (seed_extra_network_interfaces_manila if (kolla_enable_manila | bool and kolla_enable_manila_backend_cephfs_native | bool) else []) }}"
+seed_extra_network_interfaces: "{{ seed_extra_network_interfaces_external }} + {{ seed_extra_network_interfaces_manila if (kolla_enable_manila | bool and kolla_enable_manila_backend_cephfs_native | bool) else [] }}"
```

hmm, does this result in a proper list, or a string? I would have done it like this:
```suggestion
seed_extra_network_interfaces: "{{ seed_extra_network_interfaces_external + seed_extra_network_interfaces_manila if (kolla_enable_manila | bool and kolla_enable_manila_backend_cephfs_native | bool) else [] }}"
```

---

Although that looks like the original code. Perhaps you need to remove the trailing newline to make it parse as a list properly?
```
seed_extra_network_interfaces: >-
  "{{ seed_extra_network_interfaces_external +
  (seed_extra_network_interfaces_manila if (kolla_enable_manila | bool and kolla_enable_manila_backend_cephfs_native | bool) else []) }}"
```

---

Your first suggestion works but the second has the same error as before. I was slightly wrong in the merge comment:

> An unhandled exception occurred while templating '{{ network_interfaces | net_select_ethers | list }}'. Error was a <class 'ansible.errors.AnsibleError'>, original message: An unhandled exception occurred while templating '{{ seed_network_interfaces | unique | list }}'. Error was a <class 'ansible.errors.AnsibleError'>, original message: An unhandled exception occurred while templating '{{ (seed_default_network_interfaces +
        seed_extra_network_interfaces) | select | unique | list }}
    '. Error was a <class 'ansible.errors.AnsibleError'>, original message: Unexpected templating type error occurred on ({{ (seed_default_network_interfaces +
        seed_extra_network_interfaces) | select | unique | list }}
    ): can only concatenate list (not "str") to list

---

> hmm, does this result in a proper list, or a string?

Results in a list, though I do now see that it's a strange way of doing it

---

Ah, the problem is the quotes, which result in a string with quotes in it because it's using a literal block scalar:

```
seed_extra_network_interfaces: >-
  {{ seed_extra_network_interfaces_external +
  (seed_extra_network_interfaces_manila if (kolla_enable_manila | bool and kolla_enable_manila_backend_cephfs_native | bool) else []) }}
```

It's fine as it is now though.